src/rust-crypto/rust-crypto.ts

```diff
@@ -580,10 +582,18 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
 
     /**
      * The verification methods we offer to the other side during an interactive verification.
+     */
+    private _supportedVerificationMethods: string[] = ALL_VERIFICATION_METHODS;
+
+    /**
+     * Set the verification methods we offer to the other side during an interactive verification.
      *
      * If `undefined`, we will offer all the methods supported by the Rust SDK.
      */
-    public supportedVerificationMethods: string[] | undefined;
```

Comment:
Looks like a breaking change, any reason to not use a getter+setter?

Reply:
this isn't a public class, so it's not a breaking change.

Reply:
What makes it not public? Where is it marked as internal or experimental?

Reply:
Neither are a bunch of the things necessary to use the SDK so I don't think that's the rule. (Namely the Typed EventEmitter events which are necessary to set up listeners0.

Reply:
I mean, I always thought that *was* the definition of what was public/private, and in line with... pretty much every other javascript library on the planet? (There might be exceptions, but I'd expect them to at least be documented).

The fact that the react-sdk gut-wrenches bits of the internals is... deeply wrong. And yes, a lint rule in the react-sdk to enforce it would be great.

Reply:
> Neither are a bunch of the things necessary to use the SDK so I don't think that's the rule. (Namely the Typed EventEmitter events which are necessary to set up listeners0.

Not quite sure what you're referring to, but anything not being exported through index.js that we expect clients to be able to refer to is very definitely a bug.

Reply:
> Not quite sure what you're referring to

https://github.com/matrix-org/matrix-js-sdk/issues/3506

Reply:
yes well, that's a bug, and there's the issue to track it. Great.

The point remains that (a) if anybody has tried using rust crypto at all outside the react-sdk I will eat my hat; (b) even if it has, using `RustCrypto` or `RustVerificationRequest` directly is an application bug. I won't obfuscate this API just to support someone relying on the internals of the library in this way.

Reply:
https://github.com/matrix-org/matrix-js-sdk/pull/3574 then, if my argument isn't convincing.

Reply:
This is now an `@internal` class.